README.md

```diff
@@ -47,3 +47,297 @@ Details for those wishing to develop the library can be found on the [wiki](http
+          return executorService.getQueue().size();
+      }
+  })  
+}
+
+//or (Not implemented yet)
+queueDepth.setFunction(() -> executorService.getQueue().size());
+queueDepth.setFunction(() -> executorService.getQueue().size(), "label");
+```
+
+### Summary
+
+Summaries track the size and number of events.
+
+```java
+class YourClass {
+  static final Summary receivedBytes = Summary.build()
+     .name("requests_size_bytes").help("Request size in bytes.").register();
+  static final Summary requestLatency = Summary.build()
+     .name("requests_latency_seconds").help("Request latency in seconds.").register();
+  
+  void processRequest(Request req) {
+    Summary.Timer requestTimer = requestLatency.startTimer();
+    try {
+      // Your code here.
+    } finally {
+      receivedBytes.observe(req.size());
+      requestTimer.observeDuration();
+    }
+  }
+}
+```
+
+There are utilities for timing code and support quantiles:
+
+```java
+class YourClass {
+  static final Summary requestLatency = Summary.build()
+    .quantile(0.5, 0.05)   // Add 50th percentile (= median) with 5% tolerated error
+    .quantile(0.9, 0.01)   // Add 90th percentile with 1% tolerated error
+    .name("requests_latency_seconds").help("Request latency in seconds.").register();
+  
+  void processRequest(Request req) {
+    requestLatency.timer(() -> {
+        // Your code here.
+    });
+  }
+}
+```
+
+### Histogram
+
+Histograms track the size and number of events in buckets.
+This allows for aggregatable calculation of quantiles.
+
+```java
+class YourClass {
+  static final Histogram requestLatency = Histogram.build()
+     .name("requests_latency_seconds").help("Request latency in seconds.").register();
+
+  void processRequest(Request req) {
+    Histogram.Timer requestTimer = requestLatency.startTimer();
+    try {
+      // Your code here.
+    } finally {
+      requestTimer.observeDuration();
+    }
+  }
+}
+```
+
+The default buckets are intended to cover a typical web/rpc request from milliseconds to seconds.
+They can be overridden with the `buckets()` method on the `Histogram.Builder`.
+
+There are utilities for timing code:
+
+```java
+class YourClass {
+  static final Histogram requestLatency = Histogram.build()
+     .name("requests_latency_seconds").help("Request latency in seconds.").register();
+  
+  void processRequest(Request req) {
+    requestLatency.time(() -> {
+      // Your code here.  
+    });
+  }
+}
+```
+
+### Labels
+
+All metrics can have labels, allowing grouping of related time series.
+
+See the best practices on [naming](http://prometheus.io/docs/practices/naming/)
+and [labels](http://prometheus.io/docs/practices/instrumentation/#use-labels).
+
+Taking a counter as an example:
+
+```java
+class YourClass {
+  static final Counter requests = Counter.build()
+     .name("requests_total").help("Total requests.")
+     .labelNames("method").register();
+  
+  void processGetRequest() {
+    requests.labels("get").inc();
```

Comment:
Because the label value here is static, what you really should be doing is putting the labels() result in a class variable and then calling inc on it. This is so it gets initialised, and it's also faster.